### ISSUE TITLE
Samsung Chromebook Plus v2 XE520QAB (code name Nautilus): no hand/palm rejection

### ISSUE THREAD

**Issue by ShapeShifter499** [linuxwacom/xf86-input-wacom#317]
I have acquired a Samsung Chromebook Plus v2 XE520QAB (code name Nautilus). I have gone through the procedure of flashing Mr. Chromebox's UEFI firmware and installed Arch Linux with Gnome. 

Right now the stylus works but there is no hand/palm rejection. I can have my stylus right up against the screen and still trigger parts of the screen under my hand. Would adding a .tablet file help here or is there a bug?

I generated a sysinfo tarball and posted here
https://github.com/linuxwacom/wacom-hid-descriptors/issues/318

**Comment by ShapeShifter499** [comment]
I should add that while the built in stylus that can be popped out of the side of the laptop/chromebook doesn't have a button. Trying a S Pen Pro that can switch between Z Fold and S Pen mode or a standard Fold edition pen does work. 

The button by default seems to trigger the pen instead of the eraser. Essentially the eraser button and pen are swapped in their function. But only in Fold mode. The button does not do anything in S Pen mode.

So I think Samsung intended on using non-button pens or pens that could go into S Pen mode. But I think if the two are swapped in code maybe it can work as intended? enabling the pens with buttons to work?

**Comment by ShapeShifter499** [comment]
This issue is entirely fixed with this merged PR https://gitlab.freedesktop.org/libinput/libinput/-/merge_requests/877  This is needed for both Wayland and Xorg mode. I have tested on both display servers and found the same issue, the fix works on both. 

With the fix applied I found the following to be true for this specific device.

Using a "S Pen Pro" switched to "S Pen" mode and not "Z Fold" mode the button works fine, you just need to set up programs like Xournal to use the button. I was able to set up Xournal to switch to the eraser tool when the button is pressed.

If you try to use a pen designed for a Samsung Fold that does not have a "S Pen" switch, in my case I was testing a Fold 4 pen, the pen is detected as a eraser for some reason. Probably why they made the "S Pen Pro" that has a switch to move between the Fold and other devices, maybe it's the way the sensor is used on the Fold.

The built-in pen included with this device works as expected. There is no button for the built-in pen, only a click eject mechanism to aid in removing the pen.


Closing as solved.
